src/components/Search/SearchListWithHeader.tsx

```diff
@@ -42,39 +56,72 @@ function SearchListWithHeader({ListItem, onSelectRow, query, hash, data, searchT
+                    return;
+                }
+
+                setSelectedItems((prev) => {
+                    if (prev[item.keyForList]?.isSelected) {
+                        const {[item.keyForList]: omittedTransaction, ...transactions} = prev;
+                        return transactions;
+                    }
+                    return {...prev, [item.keyForList]: {isSelected: true, canDelete: item.canDelete, action: item.action}};
+                });
+
                 return;
             }
 
-            setSelectedItems((prev) => {
-                if (prev[item.keyForList]?.isSelected) {
-                    const {[item.keyForList]: omittedTransaction, ...transactions} = prev;
-                    return transactions;
-                }
-                return {...prev, [item.keyForList]: {isSelected: true, canDelete: item.canDelete, action: item.action}};
+            if (item.transactions.every((transaction) => selectedItems[transaction.keyForList]?.isSelected)) {
+                const reducedSelectedItems: SelectedTransactions = {...selectedItems};
+
+                item.transactions.forEach((transaction) => {
+                    delete reducedSelectedItems[transaction.keyForList];
+                });
+
+                setSelectedItems(reducedSelectedItems);
+                return;
+            }
+
+            setSelectedItems({
+                ...selectedItems,
+                ...Object.fromEntries(item.transactions.map(mapTransactionItemToSelectedEntry)),
             });
+        },
+        [selectedItems],
+    );
 
+    const openBottomModal = (item: TransactionListItemType | ReportListItemType | null) => {
+        if (!isSmallScreenWidth) {
             return;
         }
 
-        if (item.transactions.every((transaction) => selectedItems[transaction.keyForList]?.isSelected)) {
-            const reducedSelectedItems: SelectedTransactions = {...selectedItems};
+        setLongPressedItem(item);
+        setIsModalVisible(true);
+    };
 
-            item.transactions.forEach((transaction) => {
-                delete reducedSelectedItems[transaction.keyForList];
-            });
+    const turnOnSelectionMode = useCallback(() => {
+        setIsMobileSelectionModeActive?.(true);
+        setIsModalVisible(false);
+
+        if (longPressedItem) {
+            toggleTransaction(longPressedItem);
+        }
+    }, [longPressedItem, setIsMobileSelectionModeActive, toggleTransaction]);
 
-            setSelectedItems(reducedSelectedItems);
+    const closeBottomModal = useCallback(() => {
+        setIsModalVisible(false);
+    }, []);
+
+    useEffect(() => {
+        if (Object.keys(selectedItems).length !== 0) {
             return;
         }
 
-        setSelectedItems({
-            ...selectedItems,
-            ...Object.fromEntries(item.transactions.map(mapTransactionItemToSelectedEntry)),
-        });
-    };
+        setIsMobileSelectionModeActive?.(false);
```

Comment:
I think ideally we wouldn't leave selection mode when the user unselects everything, but only when the back button is hit or when the back button in the top bar is pressed.

cc: @shawnborton 

Also, I think we are missing the topbar change in selection mode in this PR
<img width="587" alt="image" src="https://github.com/Expensify/App/assets/5201282/5e238252-3e89-4f96-95f4-b07be91bedcc">

Reply:
cc @Expensify/design for thoughts

I could go either way... I can see where it makes sense that if you explicitly enter select multiple mode, you shouldn't leave select multiple mode unless you explicitly hit the back button. That being said, it doesn't feel unnatural for me that when you deselect everything, you leave this mode.

Reply:
I could also go either way, but I'm leaning towards only leaving select mode when you explicitly hit the back button. That does mean that it would be possible for the bulk actions button so say `0 selected` which is a bit weird, but I'm just imagining trying to manipulate my selection and getting frustrated if I briefly had zero items selected and it booted me out of select mode.

Reply:
Agree that it makes sense to leave select mode only when hitting the back button 👍  I think the alternative would feel jarring